modules/acquia_cms_headless/acquia_cms_headless.install

```diff
@@ -33,3 +39,49 @@ function acquia_cms_headless_requirements($phase) {
 
   return $requirements;
 }
+
+/**
+ * Implements hook_update_N().
+ *
+ * Update all consumers and assign content model roles and add
+ * provide `issue subrequests` permission to authenticated & anonymous role.
+ */
+function acquia_cms_headless_update_8001() {
```

Comment:
This should probably be a post_update hook. These updates are executed after all hook_update_n which ensures that Drupal is already fully repaired which makes it safe to update the entities.

Reply:
@lauriii We are not updating config entities, we are updating content entity (i.e Headless user), Also, we are adding new permission to role config entity (permission config to role entity is provide by CORE), so I think hook_update_N() should be fine.

Reply:
I believe content entities also should be updated in a post_update for the same reasons as config entities. This _could_ be fine but post_update should be safer.

Reply:
That depends on which modules and module versions a site is running, and which updates needs to be run. Like I said, this _could_ be fine but if you want to be sure, you should use post_update for this. The hook_update_n is supposed to be used for database schema changes and other low level changes like that, which are critical for the site to function. Updating entities doesn't fall into that category.

Reply:
For now, I'm merging this as this needs to be released ASAP. Will look later in detail and will see if this needs to be updated.